### ISSUE TITLE
Package.json - Empty licence breaks install

### ISSUE THREAD

**Issue by esphen** [yarnpkg/yarn#646]
**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**
When running `yarn` in a folder with 

**If the current behavior is a bug, please provide the steps to reproduce.**
```
mkdir test
cd test
yarn init # Yes to all
vim package.json 
# Replace   "license": "MIT" with   "license": ""
% yarn                                                                                
yarn install v0.15.1
info No lockfile found.
error TypeError: test@1.0.0: Cannot read property 'toUpperCase' of undefined
    at Array.<anonymous> (/home/espen/.nvm/versions/node/v6.6.0/lib/node_modules/yarnpkg/node_modules/spdx-correct/index.js:82:23)
    at validTransformation (/home/espen/.nvm/versions/node/v6.6.0/lib/node_modules/yarnpkg/node_modules/spdx-correct/index.js:173:36)
    at module.exports (/home/espen/.nvm/versions/node/v6.6.0/lib/node_modules/yarnpkg/node_modules/spdx-correct/index.js:215:21)
    at module.exports (/home/espen/.nvm/versions/node/v6.6.0/lib/node_modules/yarnpkg/node_modules/validate-npm-package-license/index.js:60:23)
    at isValidLicense (/home/espen/.nvm/versions/node/v6.6.0/lib/node_modules/yarnpkg/lib/util/normalize-manifest/util.js:16:10)
    at exports.default (/home/espen/.nvm/versions/node/v6.6.0/lib/node_modules/yarnpkg/lib/util/normalize-manifest/validate.js:53:55)
    at /home/espen/.nvm/versions/node/v6.6.0/lib/node_modules/yarnpkg/lib/util/normalize-manifest/index.js:56:51
    at next (native)
    at step (/home/espen/.nvm/versions/node/v6.6.0/lib/node_modules/yarnpkg/node_modules/babel-runtime/helpers/asyncToGenerator.js:17:30)
    at /home/espen/.nvm/versions/node/v6.6.0/lib/node_modules/yarnpkg/node_modules/babel-runtime/helpers/asyncToGenerator.js:28:20
info Visit http://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

**What is the expected behavior?**
No stack trace, install process does not break

**Please mention your node.js, yarn and operating system version.**
```
% node --version                                                                      
v6.6.0
% yarn --version                                                                      
0.15.1
OS: Gentoo fully up to date as of 11.10.2016
```

**Comment by JesseObrien** [comment]
It looks like *any* empty string in the file will break the `yarn` command.

**Comment by samccone** [comment]
This was fixed in d3a1d85fde2474ffa023041988403669bdf02f8a

Thanks!
